File: dsa-practice/arrays/codes/problemset1/prob2.cpp
#include<iostream>
using namespace std;
int main(){
    int n=5;
    int arr[n] = {2,4,5,2,3};
    for (int i = 0; i < n; i++)
    {
        int sum = 0;
        for(int j=i;j<n;j++){
            sum += arr[j];
            cout<<sum<<endl;
            

        }
        
    }
    

    return 0;
}

File: dsa-practice/arrays/codes/problemset1/prob2.ts
const sumOfSubArray = (arr:number[])=>{
    const res:number[] = [];
    arr.forEach((e,i)=>{
        let sum = 0;
        for(let j=i;i<arr.length;j++){
            sum += arr[j]
        }
        res.push(sum)
    })
    return res;
}
console.log(sumOfSubArray([2,4,5,2,3]))